### ISSUE TITLE
Avoid clash between ls dependencies subcommands and targets

### ISSUE THREAD

**Issue by mboes** [commercialhaskell/stack#5093]
Since #4424, `stack ls dependencies` has subcommands (`tree`, `text` and `json`). However, targets can also be given to `stack ls dependencies`. Say I want to list only the dependencies of the `text` package. With the latest release of Stack (v2.1.3), this is possible. But how will this work in the next version?

I would suggest an alternative CLI. Something like:

```
$ stack ls dependencies --format json
$ stack ls dependencies --format text
$ stack ls dependencies --format tree
```

**Comment by mboes** [comment]
cc  @akshaymankar

**Comment by akshaymankar** [comment]
I agree, it would be really annoying if one has to say `stack ls dependencies text text` to get dependencies of `text`.

I might not get time to do it this month, but https://github.com/commercialhaskell/stack/pull/4424#issuecomment-445003174 might be useful in deciding the implementation strategy. 

Also, as mentioned in https://github.com/commercialhaskell/stack/pull/4424#issuecomment-543310169 if the options remain `--text`, `--tree` and `--json`, it might even be possible to release this feature in a minor version.

**Comment by mboes** [comment]
I'd be happy if we stuck with `--text`, `--tree` and `--json`. I don't see any particular downside to them, and as you point out, it's what's there already, so no backwards compatibility breakage.

**Comment by mpilgrem** [comment]
I am closing this issue given the passage of time and because the `stack ls dependencies` command is now established and, while it is possible to imagine projects with packages named `text` (the name of a GHC boot package), `cabal`, `tree` or `json` (there is a package on Hackage with that name), no user of such projects has reported any actual inconvenience.
